store/sqlstore/group_supplier.go

```diff
@@ -0,0 +1,877 @@
+		} else {
+			result.Err = model.NewAppError("SqlGroupStore.GroupGetGroupSyncable", "store.select_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+		return result
+	}
+
+	result.Data = groupSyncable
+
+	return result
+}
+
+func (s *SqlSupplier) getGroupSyncable(groupID string, syncableID string, syncableType model.GroupSyncableType) (*model.GroupSyncable, error) {
+	var err error
+	var result interface{}
+
+	switch syncableType {
+	case model.GroupSyncableTypeTeam:
+		result, err = s.GetMaster().Get(GroupTeam{}, groupID, syncableID)
+	case model.GroupSyncableTypeChannel:
+		result, err = s.GetMaster().Get(GroupChannel{}, groupID, syncableID)
+	default:
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if result == nil {
+		return nil, sql.ErrNoRows
+	}
+
+	groupSyncable := model.GroupSyncable{}
+	switch syncableType {
+	case model.GroupSyncableTypeTeam:
+		groupTeam := result.(*GroupTeam)
+		groupSyncable.SyncableId = groupTeam.TeamId
+		groupSyncable.GroupId = groupTeam.GroupId
+		groupSyncable.CanLeave = groupTeam.CanLeave
+		groupSyncable.AutoAdd = groupTeam.AutoAdd
+		groupSyncable.CreateAt = groupTeam.CreateAt
+		groupSyncable.DeleteAt = groupTeam.DeleteAt
+		groupSyncable.UpdateAt = groupTeam.UpdateAt
+		groupSyncable.Type = syncableType
+	case model.GroupSyncableTypeChannel:
+		groupChannel := result.(*GroupChannel)
+		groupSyncable.SyncableId = groupChannel.ChannelId
+		groupSyncable.GroupId = groupChannel.GroupId
+		groupSyncable.CanLeave = groupChannel.CanLeave
+		groupSyncable.AutoAdd = groupChannel.AutoAdd
+		groupSyncable.CreateAt = groupChannel.CreateAt
+		groupSyncable.DeleteAt = groupChannel.DeleteAt
+		groupSyncable.UpdateAt = groupChannel.UpdateAt
+		groupSyncable.Type = syncableType
+	default:
+		return nil, fmt.Errorf("unable to convert syncableType: %s", syncableType.String())
+	}
+
+	return &groupSyncable, nil
+}
+
+func (s *SqlSupplier) GroupGetAllGroupSyncablesByGroup(ctx context.Context, groupID string, syncableType model.GroupSyncableType, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	args := map[string]interface{}{"GroupId": groupID}
+
+	appErrF := func(msg string) *model.AppError {
+		return model.NewAppError("SqlGroupStore.GroupGetAllGroupSyncablesByGroup", "store.select_error", nil, msg, http.StatusInternalServerError)
+	}
+
+	groupSyncables := []*model.GroupSyncable{}
+
+	switch syncableType {
+	case model.GroupSyncableTypeTeam:
+		sqlQuery := `
+			SELECT
+				GroupTeams.*, 
+				Teams.DisplayName AS TeamDisplayName, 
+				Teams.Type AS TeamType
+			FROM 
+				GroupTeams
+				JOIN Teams ON Teams.Id = GroupTeams.TeamId
+			WHERE 
+				GroupId = :GroupId AND GroupTeams.DeleteAt = 0`
+
+		results := []*GroupTeamJoin{}
+		_, err := s.GetMaster().Select(&results, sqlQuery, args)
+		if err != nil {
+			result.Err = appErrF(err.Error())
+			return result
+		}
+		for _, result := range results {
+			groupSyncable := &model.GroupSyncable{
```

Comment:
0/5 but since this is done in a few places, you could have a utility function that converts a `GroupTeam` to a `GroupSyncable` and similar for a `GroupChannel`.

Reply:
Good idea. Done.